debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresChangeRecordEmitter.java

```diff
@@ -88,6 +91,12 @@ public void emitChangeRecords(DataCollectionSchema schema, Receiver receiver) th
         super.emitChangeRecords(schema, receiver);
     }
 
+    @Override
+    protected void emitTruncateRecord(Receiver receiver, TableSchema tableSchema) throws InterruptedException {
+        Struct envelope = tableSchema.getEnvelopeSchema().truncate(getOffset().getSourceInfo(), getClock().currentTimeAsInstant());
+        receiver.changeRecord(tableSchema, Operation.TRUNCATE, null, envelope, getOffset(), null);
```

Comment:
How about using the table id as the message key? You still wouldn't be sure to not receive a change event from before the TRUNCATE only after that "t" event from a partitioned Kafka topic, but it wouldn't do harm either and add a tad of descriptiveness. I suppose it's no problem for you to begin with as you use the embedded engine.

Reply:
I think adding the table id as key for truncate event would be bit confusing because actual row pk is added as key for other events. If it's okay, then I could add the table id as key.

Reply:
Yes, you are probably right. After all, the table name might (unlikely though) collide with a PK value. Left's leave it null then.